targets/zephyr/src/main-zephyr.c

```diff
@@ -158,6 +158,7 @@ const struct shell_cmd commands[] =
 void main (void)
 {
   printf ("Jerry Compilation " __DATE__ " " __TIME__ "\n");
+  jerry_init (JERRY_INIT_EMPTY);
   shell_register_app_cmd_handler (shell_cmd_handler);
   shell_init ("js> ", commands);
 } /* main */
```

Comment:
@LaszloLango : That's not the case with Zephyr. Generally, a typical embedded app would work as an infinite loop (it "exits" via hardware reset), and specifically with Zephyr, shell_init() just sets up background task and exit, all actual input passed to callbacks at later time. So, it's not ok to deinitialize Jerry after call to shell_init(), and per above, no need to deinitialize at all. Hope that makes sense. I'm adding a comment to the patch with this info.